nistats/second_level_model.py

```diff
@@ -0,0 +1,451 @@
+            The resulting second level design matrix will use the same column
+            names as in the given DataFrame for confounds. At least two columns
+            are expected, "subject_id" and at least one confounder.
+
+        design_matrix: pandas DataFrame, optional
+            Design matrix that will be used to fit the GLM. The number of rows
+            in the design matrix must agree with the size of the list in
+            first_level_conditions. Careful that the order of maps given
+            by first_level_conditions or inferred directly from
+            first_level_input when first_level_conditions is None correctly
+            match the order of the rows in the design matrix.
+
+        first_level_conditions_name: list of str or None,
+            This argument only applies when first_level_input is a list of
+            `FirstLevelModel` objects. It specifies the name of the columns
+            corresponding to the provided first_level_conditions.
+            If None then the column names will be generated with the form
+            'con_00'.
+        """
+        # Check parameters
+        # check first level input
+        if isinstance(first_level_input, FirstLevelModel):
+            raise ValueError('A second level model requires a list with at'
+                             'least two first level models.')
+        elif isinstance(first_level_input, list):
+            if len(first_level_input) < 2:
+                raise ValueError('A second level model requires a list with at'
+                                 'least two first level models.')
+            # Check FirstLevelModel objects case
+            if isinstance(first_level_input[0], FirstLevelModel):
+                if first_level_conditions is None:
+                    raise ValueError('First level models input require'
+                                     'first_level_conditions to be provided')
+                for midx, first_level_model in enumerate(first_level_input):
+                    if not isinstance(first_level_model, FirstLevelModel):
+                        raise ValueError(' object at idx %d is %s instead of'
+                                         ' FirstLevelModel object' %
+                                         (midx, type(first_level_model)))
+                    if confounds is not None:
+                        if first_level_model.subject_id is None:
+                            raise ValueError('In case confounds are provided '
+                                             'first level objects need to '
+                                             'provide the attribute subject_id'
+                                             ' to match rows appropriately'
+                                             '. model at idx %d do not provide'
+                                             ' it. It can be set by doing '
+                                             'model.subject_id = "01"')
```

Comment:
first_level_model.subject_id = "01".

nitpicking: could you make longer lines by starting a new line after the opening parenthesis ?